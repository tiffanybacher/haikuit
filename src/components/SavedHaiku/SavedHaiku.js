import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Haiku from '../Haiku/Haiku';

export class SavedHaiku extends Component {
  render() {
    const id = parseInt(this.props.match.params.id);
    const haiku = this.props.haikus.find(haiku => {
      return haiku.id === id;
    });

    return (
      <section className="SavedHaiku">
        <h2>Your Haiku is Saved!</h2>
        <Haiku { ...haiku } />
        <div className="btn-container">
          <button><Link to={`/haiku/${id}/edit`}>Edit</Link></button>
          <button><Link to={'/haikus'}>See All Haikus</Link></button>
        </div>
      </section>
    );
  }
}

export const mapStoreToProps = (store) => ({
  haikus: store.haikus
});

export default connect(mapStoreToProps)(SavedHaiku);