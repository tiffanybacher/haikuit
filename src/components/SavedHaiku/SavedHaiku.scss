.SavedHaiku {
  background: white;
  padding: 20px;

  h2 {
    margin-top: 0;
    text-align: center;
  }

  .btn-container {
    display: flex;
    flex-direction: column;
  }

  button {
    background: $purple;
    color: white;
    font-weight: 600;
    margin: 10px 0;
    padding: 10px;
    width: 100%;

    &:hover {
      background: rgba($purple, .7);
    }
  }

  a {
    color: white;
    text-decoration: none;
  }

  .edit-btn {
    padding: 10px 115px;
  }

  .see-all-btn {
    padding: 10px 75px;
  }
}

@media only screen and (min-width: 768px) {
  .SavedHaiku {
    font-size: 1.1em;
    padding: 30px 50px;

    h2 {
      text-align: left;
    }

    .btn-container {
      flex-direction: row;
      justify-content: space-between;
    }

    button {
      width: 45%;
    }
  }
}

@media only screen and (min-width: 1000px) {
  .SavedHaiku {
    font-size: 1.2em;
    padding: 40px 75px;

    .edit-btn {
      padding: 10px 160px;
    }

    .see-all-btn {
      padding: 10px 115px;
    }
  }
}

@media only screen and (min-width: 1200px) {
  .SavedHaiku {
    padding: 50px 100px;

    .edit-btn {
      padding: 10px 190px;
    }

    .see-all-btn {
      padding: 10px 145px;
    }
  }
}