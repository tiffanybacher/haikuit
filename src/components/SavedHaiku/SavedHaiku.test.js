import React from 'react';
import { shallow } from 'enzyme';
import { SavedHaiku, mapStoreToProps } from './SavedHaiku';

describe('SavedHaiku', () => {
  let wrapper;
  let mockMatch;
  let mockHaikus;

  beforeEach(() => {
    mockMatch = { params: {id: 111} };
    mockHaikus = [{ title: 'Title', id: 123 }];
    wrapper = shallow(
      <SavedHaiku 
        match={mockMatch}
        haikus={mockHaikus} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map haikus from store to props', () => {
    const mockState = { haikus: [{ title: 'Boop' }] };   
    const result = mapStoreToProps(mockState);
    const expected = { haikus: mockState.haikus };

    expect(result).toEqual(expected);
  });
});