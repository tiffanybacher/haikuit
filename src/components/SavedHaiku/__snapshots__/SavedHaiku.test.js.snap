// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedHaiku should match snapshot 1`] = `
<section
  className="SavedHaiku"
>
  <Link
    className="back-home-btn"
    to="/"
  >
    Back Home
  </Link>
  <h2>
    Your Haiku is Saved!
  </h2>
  <Haiku />
  <div
    className="btn-container"
  >
    <button>
      <Link
        className="edit-btn"
        to="/haiku/111/edit"
      >
        Edit
      </Link>
    </button>
    <button>
      <Link
        className="see-all-btn"
        to="/haikus"
      >
        See All Haikus
      </Link>
    </button>
  </div>
</section>
`;
